File: src/common/logger.ts
export type LogFunctionType = (message: string, ...optionalParams: unknown[]) => void;

export enum LoggerLevelColours {
  INFO = '\x1b[36m',
  SUCCESS = '\x1b[38;2;0;128;0m',
  ERROR = '\x1b[31m',
  WARNING = '\x1b[33m',
}

/**
 *  A function to build a log message with the lens node footprint.
 * @param message The message
 */
export const buildLensNodeFootprintLog = (message: string): string => {
  return `[${new Date().toUTCString()}] LENS VERIFICATION NODE - ${message}`;
};

/**
 * A function to log a message to the console with a info font color.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleLog = (message: string, ...optionalParams: unknown[]): void => {
  console.log(
    LoggerLevelColours.INFO,
    `[${new Date().toUTCString()}] ${message}`,
    ...optionalParams
  );
};

/**
 * A function to log a message to the console with a info font color including the lens node footprint.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleLogWithLensNodeFootprint = (
  message: string,
  ...optionalParams: unknown[]
): void => {
  console.log(LoggerLevelColours.INFO, buildLensNodeFootprintLog(message), ...optionalParams);
};

/**
 * A function to log a message to the console with a green font color.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleSuccess = (message: string, ...optionalParams: unknown[]): void => {
  console.log(
    LoggerLevelColours.SUCCESS,
    `[${new Date().toUTCString()}] ${message}`,
    ...optionalParams
  );
};

/**
 * A function to log a error message to the console with a red font color.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleError = (message: string, ...optionalParams: unknown[]): void => {
  console.log(
    LoggerLevelColours.ERROR,
    `[${new Date().toUTCString()}] ${message}`,
    ...optionalParams
  );
};

/**
 * A function to log a warning message to the console with a yellow font color.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleWarn = (message: string, ...optionalParams: unknown[]): void => {
  console.log(
    LoggerLevelColours.WARNING,
    `[${new Date().toUTCString()}] ${message}`,
    ...optionalParams
  );
};

/**
 * A function to log a message to the console with a dyanmic font color.
 * @param message - The message to be logged to the console.
 * @param optionalParams - An optional list of additional parameters to be logged to the console.
 */
export const consoleDynamic = (
  color: LoggerLevelColours,
  message: string,
  ...optionalParams: unknown[]
): void => {
  console.log(color, `[${new Date().toUTCString()}] ${message}`, ...optionalParams);
};

File: momoka-rs/src/logger.rs
use crate::utils::get_current_utc_string;

/// A logger for printing log messages in different colors.
pub struct Logger;

#[allow(dead_code)]
impl Logger {
    fn print_colored_message(&self, color_code: &str, content: &str) {
        println!(
            "{}LENS VERIFICATION NODE - {} - {}{}",
            color_code,
            get_current_utc_string(),
            content,
            "\x1b[0m"
        );
    }

    /// Prints an error message with the specified content in red.
    ///
    /// # Arguments
    ///
    /// * `content` - The content of the error message.
    pub fn error(&self, content: &str) {
        self.print_colored_message("\x1b[31m", content); // Red
    }

    /// Prints a warning message with the specified content in yellow.
    ///
    /// # Arguments
    ///
    /// * `content` - The content of the warning message.
    pub fn warning(&self, content: &str) {
        self.print_colored_message("\x1b[33m", content); // Yellow
    }

    /// Prints an informational message with the specified content in blue.
    ///
    /// # Arguments
    ///
    /// * `content` - The content of the informational message.
    pub fn info(&self, content: &str) {
        self.print_colored_message("\x1b[36m", content); // Blue
    }

    /// Prints a success message with the specified content in green.
    ///
    /// # Arguments
    ///
    /// * `content` - The content of the success message.
    pub fn success(&self, content: &str) {
        self.print_colored_message("\x1b[38;2;0;128;0m", content); // Green
    }
}
